spec/rspec/core/configuration_spec.rb

```diff
@@ -1229,7 +1229,167 @@ def metadata_hash(*args)
       end
     end
 
+    describe "#color_mode" do
+      context ":automatic" do
+        before do
+          config.color_mode = :automatic
+        end
+
+        context "with output.tty?" do
+          it "sets color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages(:tty? => true)
+
+            expect(config.color_enabled?).to be true
+            expect(config.color_enabled?(output)).to be true
+          end
+        end
+
+        context "with !output.tty?" do
+          it "sets !color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages(:tty? => false)
+
+            expect(config.color_enabled?).to be false
+            expect(config.color_enabled?(output)).to be false
+          end
+        end
+      end
+
+      context ":forced" do
+        before do
+          config.color_mode = :forced
+        end
+
+        context "with output.tty?" do
+          it "sets color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages(:tty? => true)
+
+            expect(config.color_enabled?).to be true
+            expect(config.color_enabled?(output)).to be true
+          end
+        end
+
+        context "with !output.tty?" do
+          it "sets color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages(:tty? => false)
+
+            expect(config.color_enabled?).to be true
+            expect(config.color_enabled?(output)).to be true
+          end
+        end
+      end
+
+      context ":disabled" do
+        before do
+          config.color_mode = :disabled
+        end
+
+        context "with output.tty?" do
+          it "sets !color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages(:tty? => true)
+
+            expect(config.color_enabled?).to be false
+            expect(config.color_enabled?(output)).to be false
+          end
+        end
+
+        context "with !output.tty?" do
+          it "sets !color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages(:tty? => false)
+
+            expect(config.color_enabled?).to be false
+            expect(config.color_enabled?(output)).to be false
+          end
+        end
+      end
+
+      describe "#color_mode=" do
+        it "prefers incoming cli_args" do
+          config.output_stream = StringIO.new
+          allow(config.output_stream).to receive_messages(:tty? => true)
```

Comment:
What does the output stream have to do with the rest of this example?  It does not seem to be used since it does not check `color_enabled?`...